cpp/src/parquet/stream_reader_test.cc

```diff
@@ -201,34 +268,34 @@ TEST_F(TestStreamReader, TypeChecking) {
   double d;
   std::string str;
 
-  ASSERT_THROW(reader_ >> int8, ParquetException);
-  ASSERT_NO_THROW(reader_ >> b);
-  ASSERT_THROW(reader_ >> c, ParquetException);
-  ASSERT_NO_THROW(reader_ >> s);
-  ASSERT_THROW(reader_ >> s, ParquetException);
-  ASSERT_NO_THROW(reader_ >> c);
-  ASSERT_THROW(reader_ >> s, ParquetException);
-  ASSERT_NO_THROW(reader_ >> char_array);
-  ASSERT_THROW(reader_ >> int16, ParquetException);
-  ASSERT_NO_THROW(reader_ >> int8);
-  ASSERT_THROW(reader_ >> int16, ParquetException);
-  ASSERT_NO_THROW(reader_ >> uint16);
-  ASSERT_THROW(reader_ >> int64, ParquetException);
-  ASSERT_NO_THROW(reader_ >> int32);
-  ASSERT_THROW(reader_ >> int64, ParquetException);
-  ASSERT_NO_THROW(reader_ >> uint64);
-  ASSERT_THROW(reader_ >> uint64, ParquetException);
-  ASSERT_NO_THROW(reader_ >> ts_us);
-  ASSERT_THROW(reader_ >> d, ParquetException);
-  ASSERT_NO_THROW(reader_ >> f);
-  ASSERT_THROW(reader_ >> f, ParquetException);
-  ASSERT_NO_THROW(reader_ >> d);
-  ASSERT_NO_THROW(reader_ >> EndRow);
+  EXPECT_THROW(reader_ >> int8, ParquetException);
+  EXPECT_NO_THROW(reader_ >> b);
+  EXPECT_THROW(reader_ >> c, ParquetException);
+  EXPECT_NO_THROW(reader_ >> s);
+  EXPECT_THROW(reader_ >> s, ParquetException);
+  EXPECT_NO_THROW(reader_ >> c);
+  EXPECT_THROW(reader_ >> s, ParquetException);
+  EXPECT_NO_THROW(reader_ >> char_array);
+  EXPECT_THROW(reader_ >> int16, ParquetException);
+  EXPECT_NO_THROW(reader_ >> int8);
+  EXPECT_THROW(reader_ >> int16, ParquetException);
+  EXPECT_NO_THROW(reader_ >> uint16);
+  EXPECT_THROW(reader_ >> int64, ParquetException);
+  EXPECT_NO_THROW(reader_ >> int32);
+  EXPECT_THROW(reader_ >> int64, ParquetException);
+  EXPECT_NO_THROW(reader_ >> uint64);
+  EXPECT_THROW(reader_ >> uint64, ParquetException);
+  EXPECT_NO_THROW(reader_ >> ts_us);
+  EXPECT_THROW(reader_ >> d, ParquetException);
+  EXPECT_NO_THROW(reader_ >> f);
+  EXPECT_THROW(reader_ >> f, ParquetException);
+  EXPECT_NO_THROW(reader_ >> d);
+  EXPECT_NO_THROW(reader_ >> EndRow);
```

Comment:
Does changing all of these to EXPECT make sense? If one of them fails then probably all of the rest are, too